contracts/fuses/stake_dao_v2/StakeDaoV2SupplyFuse.sol

```diff
@@ -0,0 +1,179 @@
+// SPDX-License-Identifier: BUSL-1.1
+pragma solidity 0.8.26;
+
+import {SafeCast} from "@openzeppelin/contracts/utils/math/SafeCast.sol";
+import {SafeERC20} from "@openzeppelin/contracts/token/ERC20/utils/SafeERC20.sol";
+import {ERC20} from "@openzeppelin/contracts/token/ERC20/ERC20.sol";
+import {IERC4626} from "@openzeppelin/contracts/interfaces/IERC4626.sol";
+import {Errors} from "../../libraries/errors/Errors.sol";
+import {IporMath} from "../../libraries/math/IporMath.sol";
+import {IFuseCommon} from "../IFuseCommon.sol";
+import {PlasmaVaultConfigLib} from "../../libraries/PlasmaVaultConfigLib.sol";
+
+struct StakeDaoV2SupplyFuseEnterData {
+    /// @dev Stake DAO V2 reward vault address, with underlying asset as lp token
+    /// @dev Example: Stake DAO Curve Vault for crvUSD Vault (sd-cvcrvUSD-vault) [0x1544E663DD326a6d853a0cc4ceEf0860eb82B287]
+    address rewardVault;
+    /// @dev amount of lp token underlying asset amount to supply,
+    /// @dev Example: Curve.Fi USD Stablecoin (crvUSD) [0x498Bf2B1e120FeD3ad3D42EA2165E9b73f99C1e5] in vault Curve Vault for crvUSD (cvcrvUSD) [0xe07f1151887b8FDC6800f737252f6b91b46b5865]
+    uint256 lpTokenUnderlyingAmount;
+    /// @dev minimum amount of lp token underlying asset to supply, if not enough underlying asset is supplied, the enter will revert
+    uint256 minLpTokenUnderlyingAmount;
+}
+
+struct StakeDaoV2SupplyFuseExitData {
+    /// @dev Stake DAO V2 reward vault address, with underlying asset as lp token
+    /// @dev Example: Stake DAO Curve Vault for crvUSD Vault (sd-cvcrvUSD-vault) [0x1544E663DD326a6d853a0cc4ceEf0860eb82B287]
+    address rewardVault;
+    /// @dev amount of reward vault shares to withdraw,
+    uint256 rewardVaultShares;
+    /// @dev minimum amount of reward vault shares to withdraw,
+    uint256 minRewardVaultShares;
+    /// @dev mode of exit,
+    /// @dev 0 - exit from reward vault,
+    /// @dev 1 - exit from reward vault AND then exit from lp token vault (LP token vault is an underlying asset of reward vault)
+    uint256 mode;
+}
+
+contract StakeDaoV2SupplyFuse is IFuseCommon {
+    using SafeCast for uint256;
+    using SafeERC20 for ERC20;
+
+    address public immutable VERSION;
+    uint256 public immutable MARKET_ID;
+
+    event StakeDaoV2SupplyFuseEnter(
+        address version,
+        address rewardVault,
+        uint256 rewardVaultShares,
+        uint256 lpTokenAmount,
+        uint256 finalLpTokenUnderlyingAmount
+    );
+    event StakeDaoV2SupplyFuseExit(
+        address version,
+        address rewardVault,
+        uint256 finalRewardVaultShares,
+        uint256 lpTokenAmount,
+        uint256 lpTokenUnderlyingAmount,
+        uint256 mode
```

Comment:
Are we sure we want this? 
It means that we need 2 priceFeeds to operate one market; 
without this, one is enough.